### PR TITLE
Refactor `node-loader` to make it ready for multiple tasks

### LINKED ISSUE #1502: Make `node-loader` ready for multiple tasks
Also logs are unreadable when multiple workers are enabled, so need to rework that.

### PR DESCRIPTION
**Pull request by SabaunT**
Resolves #1502 .

- Separate models and logic into corresponding types and modules
- Implement in-memory logger which avoids logs jumbling

### CODE REVIEW

File: gear-test/src/check.rs [hunk 1]
```diff
@@ -727,7 +727,7 @@ where
                         .unwrap()
                         .iter()
                         .for_each(|line| {
-                            println!("{}", line);
+                            println!("{}", &thread::current().id(), line);
```
Comment (SabaunT) [inline gear-test/src/check.rs:730]: redundant, will be restored

File: utils/node-loader/src/args.rs [hunk 2]
```diff
@@ -1,35 +1,72 @@
...
 }
+
+#[derive(Debug)]
+pub(crate) enum SeedVariant {
+    Dynamic(u64),
+    Constant(u64),
+}
+
+impl FromStr for SeedVariant {
+    type Err = String;
```
Comment (breathx) [inline utils/node-loader/src/args.rs:53]: I'd add anyhow error here

File: utils/node-loader/src/main.rs [hunk 3]
```diff
@@ -5,113 +5,58 @@
...
 mod args;
+mod reporter;
+mod task;
+mod utils;
 
+/// Main entry-point
 #[tokio::main]
-async fn main() -> Result<()> {
+async fn main() {
+    if let Err(e) = run().await {
```
Comment (breathx) [inline utils/node-loader/src/main.rs:25]: I'd move parsing params here in order to provide params arg to run fn

File: utils/node-loader/src/main.rs [hunk 4]
```diff
@@ -5,113 +5,58 @@
...
-        return Ok(());
+    match params {
+        Params::Dump { seed } => dump_with_seed(seed),
+        Params::Load(load_params) => load_node(load_params).await,
     }
+}
 
-    load_node(params).await;
-
+// todo use anyhow?
```
Comment (breathx) [inline utils/node-loader/src/main.rs:40]:
```suggestion
```

File: utils/node-loader/src/task/upload_program.rs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    pub(super) fn gen<Rng: utils::Rng>(
+        &mut self,
+    ) -> impl Future<Output = SomeReporter> + Send + 'static {
+        upload_program_task::<Rng>(self.gear_api.clone(), self.code_seed_gen.next_u64())
+    }
+}
+
+async fn upload_program_task<Rng: utils::Rng>(gear_api: Api, code_gen_seed: u64) -> SomeReporter {
+    // todo avoid panics
```
Comment (breathx) [inline utils/node-loader/src/task/upload_program.rs:32]: anyhow error and `?`

Comment (breathx) [reply utils/node-loader/src/task/upload_program.rs:32]: same further

**Review by breathx — APPROVED** [review]

**Pull request merged**